src/app/3d/qgspointcloud3dsymbolwidget.cpp

```diff
@@ -88,6 +87,17 @@ QgsPointCloud3DSymbolWidget::QgsPointCloud3DSymbolWidget( QgsPointCloudLayer *la
   mRenderingStyleComboBox->setCurrentIndex( 0 );
   mStackedWidget->setCurrentIndex( 0 );
 
+  mPointBudgetPercentageSlider->setMinimum( 0.0 );
+  mPointBudgetPercentageSlider->setMaximum( 100.0 );
+
+  mPointBudgetPercentage = std::min( 100.0, 100.0 * 1000000.0 / mLayer->pointCount() );
+  whileBlocking( mPointBudgetPercentageSlider ) ->setValue( mPointBudgetPercentage );
+  whileBlocking( mPointBudgetPercentageSpinBox ) ->setValue( mPointBudgetPercentage );
+
+  whileBlocking( mRenderedPointsSpinBox )->setMinimum( 0 );
+  whileBlocking( mRenderedPointsSpinBox )->setMaximum( mLayer->pointCount() + 1 );
+  whileBlocking( mRenderedPointsSpinBox )->setValue( 1000000 );
```

Comment:
Hmm I don't think a slider with percentage of points of a layer would work well. The point budget should be something based on how good user's GPU is... so the user would normally use e.g. 1M points budget irrespective from whether the point cloud layer has 10M points of 1 billion points. With the percentage slider this would be more complicated to set up - and e.g. with 1 billion points, the 1% would be 10M which could be too much for many :-)

My suggestion is to simply have there a spin box with budget. Maybe we can always work with millions, so the range of allowed values would go e.g. from 0.1 to 100 millions. (Even 100M could be too much, maybe 10M would be a more sensible limit?)

Reply:
+1 for a spin only. I'd also suggest we add a global setting in the 3d options panel for "Default point cloud budget", which is inherited by newly added point cloud layers. That way users with underpowered graphics cards could set a global default to a lower number so that adding the clouds immediately results in a usable layer for them...

Reply:
Good idea with the "default point cloud budget" option in global settings!